sklearn/datasets/tests/test_mldata.py

```diff
@@ -46,20 +48,23 @@ def test_download(tmpdata):
         },
     })
     try:
-        mock = fetch_mldata('mock', data_home=tmpdata)
+        mock = assert_warns(DeprecationWarning, fetch_mldata,
+                            'mock', data_home=tmpdata)
         for n in ["COL_NAMES", "DESCR", "target", "data"]:
             assert_in(n, mock)
 
         assert_equal(mock.target.shape, (150,))
         assert_equal(mock.data.shape, (150, 4))
 
         assert_raises(datasets.mldata.HTTPError,
+                      assert_warns, DeprecationWarning,
                       fetch_mldata, 'not_existing_name')
     finally:
         datasets.mldata.urlopen = _urlopen_ref
 
 
-def test_fetch_one_column(tmpdata):
+def test_fetch_one_column(tmpdata, recwarn):
+    warnings.simplefilter('ignore', DeprecationWarning)
```

Comment:
Maybe rather use, 
```
@pytest.mark.filterwarnings('ignore::DeprecationWarning'):
def test_fetch_one_column(tmpdata):
   ...
```
if I got the syntax right. I'm never know about the side effects explicitly setting warning filters could have in a pytest session..

Reply:
I thought recwarn grabbed a filter context... But you're right that syntax is a little better.

Reply:
It does, whether this isolates the filter context (and will continue doing so in the future) is another story. That's why using standard pytest mechanisms is probably safer (besides syntax considerations).